lib/src/util/side_effect_visitor.dart

```diff
@@ -0,0 +1,546 @@
+import 'package:analyzer/analyzer.dart';
+
+class SideEffectVisitor extends RecursiveAstVisitor {
+  var _hasSideEffect = false;
+
+  bool get hasSideEffect => _hasSideEffect;
+  @override
+  visitAdjacentStrings(AdjacentStrings node) {
+    super.visitAdjacentStrings(node);
```

Comment:
If you're doing nothing other than invoking the overridden method, then there's no reason to override the method at all.